[Bug]: entered unreachable code: Always rewrite to MemberExpression for nested MemberExpression

### Reproduction link or steps

failing GitHub Actions PR in CI
https://github.com/ghiscoding/slickgrid-universal-vite-demo/pull/5

```sh
# install and build with rolldown-vite
npm install
npm run build
```

I'm getting this error in the CI

```shell
$ tsc && vite build --mode production
rolldown-vite v7.1.10 building for production...

transforming...✓ 731 modules transformed.

thread '<unnamed>' panicked at crates/rolldown/src/module_finalizers/impl_visit_mut.rs:523:11:
internal error: entered unreachable code: Always rewrite to MemberExpression for nested MemberExpression
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
✗ Build failed in 10.55s
error during build:
internal error: entered unreachable code: Always rewrite to MemberExpression for nested MemberExpression
Error: internal error: entered unreachable code: Always rewrite to MemberExpression for nested MemberExpression
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

I don't use any Rollup or esbuild plugins, my config is pretty basic

```ts
import dns from 'dns';
import { defineConfig } from 'vite';

dns.setDefaultResultOrder('verbatim');

export default defineConfig(({ mode }) => {
  return {
    base: './',
    build: {
      chunkSizeWarningLimit: 6000,
      emptyOutDir: true,
    },
    preview: {
      port: 3000
    },
    server: {
      port: 3000,
      cors: true,
      host: 'localhost',
    },
  };
});
```

### What is expected?

no errors, it works fine with latest regular Vite (7.1.5)

### What is actually happening?

It fails on the build with rolldown-vite

### System Info

It's the GitHub Actions CI using "Ubuntu-latest" but it also fails on my local Windows 10 machine

```Shell
Ubuntu  24.04.3  LTS
Node 22.19.0
NPM 10.9.3
```

### Any additional comments?

_No response_

--- a/crates/rolldown/src/module_finalizers/impl_visit_mut.rs
+++ b/crates/rolldown/src/module_finalizers/impl_visit_mut.rs
@@ -444,6 +444,15 @@ impl<'ast> VisitMut<'ast> for ScopeHoistingFinalizer<'_, 'ast> {
           *expr = self.snippet.builder.expression_object(SPAN, self.snippet.builder.vec());
         }
       }
+      ast::Expression::ChainExpression(chain_expr) => {
+        if let Some(new_expr) = chain_expr
+          .expression
+          .as_member_expression_mut()
+          .and_then(|expr| self.try_rewrite_member_expr(expr))
+        {
+          *expr = new_expr;
+        }
+      }
       _ => {
         if let Some(new_expr) =
           expr.as_member_expression().and_then(|expr| self.try_rewrite_member_expr(expr))

--- a/crates/rolldown/tests/rolldown/issues/6231/_config.json
+++ b/crates/rolldown/tests/rolldown/issues/6231/_config.json
@@ -1,0 +1,4 @@
+{
+  "config": {},
+  "expectExecuted": false
+}

--- a/crates/rolldown/tests/rolldown/issues/6231/foo.js
+++ b/crates/rolldown/tests/rolldown/issues/6231/foo.js
@@ -1,0 +1,1 @@
+export const bar = 'bar';

--- a/crates/rolldown/tests/rolldown/issues/6231/main.js
+++ b/crates/rolldown/tests/rolldown/issues/6231/main.js
@@ -1,0 +1,2 @@
+import * as foo from './foo';
+console.log(foo?.bar)

